sklearn/tests/test_metadata_routing.py

```diff
@@ -239,6 +239,28 @@ class InvalidObject:
         process_routing(InvalidObject(), "fit", groups=my_groups)
 
 
+def test_process_routing_empty_params_get_with_default():
+    empty_params = {}
+    routed_params = process_routing(ConsumingClassifier(), "fit", **empty_params)
+
+    # Behaviour should be an empty dictionary returned for each method when retrieved.
+    for method in METHODS:
+        # This behaviour should be equivalent with using `get` with no default
+        params_for_method = routed_params[method]
+
+        # An empty dictionary for each method
+        assert isinstance(params_for_method, dict)
+        assert set(params_for_method.keys()) == set(METHODS)
+
+        default_params_for_method = routed_params.get(method)
+        assert default_params_for_method == params_for_method
+
+        # However, with a default, should return that instead.
+        assert routed_params.get(method, default="default") == "default"
+        assert routed_params.get(method, default=[]) == []
```

Comment:
The test is failing but I don't think that we are using the sentinel in the file so it looks normal :)

Reply:
Whoops, never actually committed that change. I'm doing this PR in between little bits of free time. Apologies for the mini mistakes